subscriptions/tasks.py

```diff
@@ -182,19 +182,13 @@ def run(self, subscription_id, **kwargs):
                 # If next set defined create new subscription
                 messageset = subscription.messageset
                 if messageset.next_set:
-                    # clone existing minus PK as recommended in
-                    # https://docs.djangoproject.com/en/1.9/topics/db/queries/
-                    # copying-model-instances
-                    subscription.pk = None
-                    subscription.process_status = 0  # Ready
-                    subscription.active = True
-                    subscription.completed = False
-                    subscription.next_sequence_number = 1
-                    subscription = subscription
-                    subscription.message_set = messageset.next_set
-                    subscription.schedule = (
-                        subscription.messageset.default_schedule)
-                    subscription.save()
+                    newsub = Subscription.objects.create(
+                        identity=subscription.identity,
+                        lang=subscription.lang,
+                        messageset=messageset.next_set,
+                        schedule=subscription.messageset.default_schedule
```

Comment:
I think this is pointing at the previous messagest schedule, rather than the new one.